amazon-kinesis-client-multilang/src/main/java/software/amazon/kinesis/multilang/config/KinesisClientLibConfigurator.java

```diff
@@ -69,7 +71,30 @@ public MultiLangDaemonConfiguration getConfiguration(Properties properties) {
         });
 
         Validate.notBlank(configuration.getApplicationName(), "Application name is required");
-        Validate.notBlank(configuration.getStreamName(), "Stream name is required");
+
+        if (configuration.getStreamArn() != null && !configuration.getStreamArn().trim().isEmpty()) {
+            Arn streamArnObj = Arn.fromString(configuration.getStreamArn());
```

Comment:
Nit: +`final` (and elsewhere this could apply)

This helps signal -- both to the JDK and future maintainers, with emphasis towards human benefit -- "this assignment isn't going to change".